### PR TITLE
Networking: some Ethernet cleanups

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

File: include/net/ethernet.h [hunk 1]
```diff
@@ -671,7 +671,16 @@ int net_eth_promisc_mode(struct net_if *iface, bool enable);
  * @return Pointer to PTP clock if found, NULL if not found or if this
  * ethernet interface does not support PTP.
  */
+#if defined(CONFIG_PTP_CLOCK)
```
Comment (jukkar) [inline include/net/ethernet.h]: I have this modification already done as part of #16403

Comment (tbursztyka) [reply include/net/ethernet.h]: ah yes, merge your PR I'll rebase

**Comment by pfalcon** [comment]
@tbursztyka, needs rebase.

File: subsys/net/l2/ethernet/lldp/Kconfig [hunk 2]
```diff
@@ -210,4 +193,4 @@ config NET_LLDP_PORT_ID
 	default "PORT_ID_PLACEHOLDER"
 endif
 
-endmenu
+endif # NET_LLDP
```
Comment (jukkar) [inline subsys/net/l2/ethernet/lldp/Kconfig]: It looks like the Kconfig file is missing newline

**Review by jukkar — CHANGES_REQUESTED** [review]
One newline was missing from Kconfig file, looks good otherwise.

**Review by jukkar — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**